src/main/java/com/spotify/docker/client/DockerHost.java

```diff
@@ -17,52 +17,73 @@
 
 package com.spotify.docker.client;
 
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.net.HostAndPort;
 
 import java.net.URI;
+import java.nio.file.Paths;
 import java.util.Locale;
 
-import static com.google.common.base.Optional.fromNullable;
+import static com.google.common.base.MoreObjects.firstNonNull;
 import static com.google.common.base.Strings.isNullOrEmpty;
-import static java.lang.System.getenv;
 
 /**
  * Represents a dockerd endpoint. A codified DOCKER_HOST.
  */
 public class DockerHost {
 
-  public static final int DEFAULT_PORT = 2375;
-  public static final String DEFAULT_HOST = "localhost";
-  public static final String DEFAULT_UNIX_ENDPOINT = "unix:///var/run/docker.sock";
+  /**
+   * An interface to be mocked during testing.
+   */
+  @VisibleForTesting
+  interface SystemDelegate {
+    String getProperty(String key);
+    String getenv(String name);
+  }
+
+  private static SystemDelegate systemDelegate = new SystemDelegate() {
```

Comment:
What is here feels like the best thing that I know of for this case. I suppose it is worth minimizing the chain of static method -> static method calls because of this reason.